File: backend/employeeManagement/src/main/java/com/example/employeeManagement/services/EmployeeService.java
package com.example.employeeManagement.services;

import com.example.employeeManagement.models.Employees;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface EmployeeService {

    List<Employees> getAllEmployees() throws Exception;
    Employees getEmployeeById(int empId) throws Exception;
    Employees addEmployee(Employees employee) throws Exception;
    Employees updateEmployee(Employees employee, int empId) throws Exception;
    boolean deleteEmployeeById(int empId) throws Exception;
}

File: frontend/src/services/employeeService.js
import axios from 'axios';

const EMPLOYEE_API_BASE_URL = "http://localhost:8080/api/employees";

class EmployeeService {

    getAllEmployees() {
        return axios.get(EMPLOYEE_API_BASE_URL + "/getAllEmployees")
    }

    getEmployeeById(empId) {
        return axios.get(EMPLOYEE_API_BASE_URL + "/getEmployeeById/" + empId)
    }

    addEmployee(employee) {
        return axios.post(EMPLOYEE_API_BASE_URL + "/addEmployee", employee)
    }

    updateEmployee(empId, employee) {
        return axios.put(EMPLOYEE_API_BASE_URL + "/updateEmployee/" + empId, employee)
    }

    delete(empId) {
        return axios.delete(EMPLOYEE_API_BASE_URL + "/delete/" + empId)
    }

}

export default new EmployeeService();